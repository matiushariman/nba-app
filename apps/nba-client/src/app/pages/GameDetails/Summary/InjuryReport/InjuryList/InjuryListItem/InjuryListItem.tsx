import PlayerHeadshot from '../../../../../../components/PlayerHeadshot';
import TeamLogo from '../../../../../../components/TeamLogo';

import type { InjuryListItemProps } from './InjuryListItem.types';

const InjuryListItem = ({
  personId,
  name,
  teamId,
  teamTricode,
  jerseyNum,
}: InjuryListItemProps) => (
  <div className="pt-2 pb-2 flex items-center gap-2">
    <div>
      <PlayerHeadshot personId={personId} name={name} size="md" />
    </div>
    <div>
      <TeamLogo teamId={teamId} teamName={teamTricode} width={24} height={24} />
    </div>
    <div className="grid">
      <p className="text-sm">{name}</p>
      <p className="text-xs text-gray-500">
        {teamTricode} | #{jerseyNum}
      </p>
    </div>
  </div>
);

export default InjuryListItem;
