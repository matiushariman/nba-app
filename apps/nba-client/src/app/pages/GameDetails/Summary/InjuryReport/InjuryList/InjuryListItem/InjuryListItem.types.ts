import type { BoxscorePlayer, Team } from '@nba-app/types-nba';

export interface InjuryListItemProps
  extends Pick<BoxscorePlayer, 'personId' | 'name' | 'jerseyNum'> {
  readonly teamId: Team['teamId'];
  readonly teamTricode: Team['teamTricode'];
}
