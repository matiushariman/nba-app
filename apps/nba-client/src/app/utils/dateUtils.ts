import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
