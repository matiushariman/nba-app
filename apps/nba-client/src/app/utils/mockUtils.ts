export const shouldUseMock = process.env['NODE_ENV'] === 'development';
