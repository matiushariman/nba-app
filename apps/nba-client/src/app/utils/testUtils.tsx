import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
