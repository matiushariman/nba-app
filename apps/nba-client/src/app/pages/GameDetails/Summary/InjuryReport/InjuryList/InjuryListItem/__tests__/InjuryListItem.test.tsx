import InjuryListItem from '..';
import { render } from '../../../../../../../utils/testUtils';

describe('pages/GameDetails/Summary/InjuryReport/InjuryList/InjuryListItem', () => {
  it('should render InjuryListItem that matches snapshot', () => {
    const { baseElement } = render(
      <InjuryListItem
        teamId={1}
        teamTricode="BOS"
        personId={1}
        jerseyNum="0"
        name="Jayson Tatum"
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
