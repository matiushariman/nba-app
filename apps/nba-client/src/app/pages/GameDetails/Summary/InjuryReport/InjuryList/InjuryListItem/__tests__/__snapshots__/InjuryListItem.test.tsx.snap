// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/Summary/InjuryReport/InjuryList/InjuryListItem should render InjuryListItem that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="pt-2 pb-2 flex items-center gap-2"
    >
      <div>
        <div
          class="w-[48px] h-[48px]"
        >
          <img
            alt="Jayson Tatum headshot"
            class="mr-2 bg-gray-300 shadow-sm rounded-full w-full h-full object-cover"
            loading="lazy"
            src="https://cdn.nba.com/headshots/nba/latest/260x190/1.png"
          />
        </div>
      </div>
      <div>
        <img
          alt="BOS logo"
          height="24"
          loading="lazy"
          src="https://cdn.nba.com/logos/nba/1/global/L/logo.svg"
          width="24"
        />
      </div>
      <div
        class="grid"
      >
        <p
          class="text-sm"
        >
          Jayson Tatum
        </p>
        <p
          class="text-xs text-gray-500"
        >
          BOS
           | #
          0
        </p>
      </div>
    </div>
  </div>
</body>
`;
